### PR TITLE
BART models

### PR DESCRIPTION
**Pull request by topepo**
Tests will be in the extratests package.

### CODE REVIEW

File: R/bart.R [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+#' bart(mode = "regression", trees = 5)
+#'
+#' # ------------------------------------------------------------------------------
+#' # Examples for terminal node prior
+#'
+#' library(ggplot2)
+#' library(dplyr)
+#'
+#' prior_test <- function(coef = 0.95, expo = 2, depths = 1:10) {
+#'   tidyr::crossing(coef = coef, expo = expo, depth = depths)  %>%
```
Comment (EmilHvitfeldt) [inline R/bart.R:53]:
```suggestion
#'   tidyr::crossing(coef = coef, expo = expo, depth = depths) %>%
```

File: R/bart.R [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+#'
+#' @seealso \Sexpr[stage=render,results=rd]{parsnip:::make_seealso_list("bart")}
+#'
+#' @examples
+#' show_engines("bart")
+#'
+#' bart(mode = "regression", trees = 5)
+#'
+#' # ------------------------------------------------------------------------------
+#' # Examples for terminal node prior
```
Comment (EmilHvitfeldt) [inline R/bart.R:47]: Could you add a line explaining why this example is being shown here?

File: R/bart.R [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+#' @export
+bart <-
+  function(mode = "unknown", engine = "dbarts",
+           trees = NULL, prior_terminal_node_coef = NULL,
+           prior_terminal_node_expo = NULL,
+           prior_outcome_range = NULL) {
+
+    args <- list(
+      trees    = enquo(trees),
```
Comment (EmilHvitfeldt) [inline R/bart.R:74]:
```suggestion
      trees = enquo(trees),
```

File: R/bart.R [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+      prior_terminal_node_coef  = enquo(prior_terminal_node_coef),
```
Comment (EmilHvitfeldt) [inline R/bart.R:75]:
```suggestion
      prior_terminal_node_coef = enquo(prior_terminal_node_coef),
```

File: R/bart.R [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+      prior_terminal_node_expo = enquo(prior_outcome_range),
+      prior_outcome_range = enquo(prior_terminal_node_expo)
```
Comment (EmilHvitfeldt) [inline R/bart.R:77]: This should be swapped right?

```suggestion
      prior_terminal_node_expo = enquo(prior_terminal_node_expo),
      prior_outcome_range = enquo(prior_outcome_range)
```

File: R/bart.R [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+           fresh = FALSE, ...) {
+
+    eng_args <- update_engine_parameters(object$eng_args, ...)
+
+    if (!is.null(parameters)) {
+      parameters <- check_final_param(parameters)
+    }
+
+    args <- list(
+      trees    = enquo(trees),
```
Comment (EmilHvitfeldt) [inline R/bart.R:124]:
```suggestion
      trees = enquo(trees),
```

File: R/bart.R [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+      prior_terminal_node_coef  = enquo(prior_terminal_node_coef),
```
Comment (EmilHvitfeldt) [inline R/bart.R:125]:
```suggestion
      prior_terminal_node_coef = enquo(prior_terminal_node_coef),
```

**Review by EmilHvitfeldt — APPROVED** [review]
LGTM!

File: man/rmd/bart_dbarts.Rmd [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+This model has `r nrow(param)` tuning parameters:
+
+```{r bart-param-list, echo = FALSE, results = "asis"}
+param$item
+```
+
+## Important engine-specific options
+
+Some relevant arguments that can be passed to `set_engine()`:
```
Comment (hfrick) [inline man/rmd/bart_dbarts.Rmd:41]: is this missing the options?

File: man/rmd/bart_dbarts.Rmd [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  set_mode("regression") %>% 
+  translate()
+```
+
+## Preprocessing requirements
+
+```{r child = "template-makes-dummies.Rmd"} 
+```
+
+[dbarts::bart()] will also convert the factors to indicators of the user does not create them first.
```
Comment (hfrick) [inline man/rmd/bart_dbarts.Rmd:78]:
```suggestion
[dbarts::bart()] will also convert the factors to indicators if the user does not create them first. 
```

**Review by hfrick — APPROVED** [review]
LGTM! Main question: are the dials objects meant to be in this PR?

**Pull request merged**